src/data/SeriesUtils.tsx

```diff
@@ -308,12 +328,19 @@ function buildSeries(
           aggregated = sum;
           break;
         case 'avg':
-          aggregated = sum / binSize;
+          if (zeroType == 'empty'){
+            // do not include nulls in average calculation
+            aggregated = sum / (binSize - nullCount);
```

Comment:
bit of an edge case, but maybe add a check for the case where all values are null/zero (set aggregated to 0, or raise some descriptive error)

Reply:
Sounds good, handled that case.